### ISSUE TITLE
CirceJsonMapper doesn't support json integers large than MaxLong

### ISSUE THREAD

**Issue by cwholmes** [Kirill5k/mongo4cats#37]
The CirceJsonMapper checks for decimal values by the existence of a decimal and then tries to convert all values without a decimal to an integer (int32) or long (int64). This fails for the case that a json integer is provided that exceeds the 64 bit long value.

Example that will fail:

```scala
import mongo4cats.bson.syntax._
import mongo4cats.circe._
import io.circe.Json

Json.obj("decimal_value" -> Json.fromBigInt(BigInt(Long.MaxValue) + 1)).toBson
```

With the following

```
java.util.NoSuchElementException: None.get
  scala.None$.get(Option.scala:627)
  scala.None$.get(Option.scala:626)
  mongo4cats.circe.CirceJsonMapper$JsonNumberSyntax$.toBsonValue$extension(CirceJsonMapper.scala:67)
  mongo4cats.circe.CirceJsonMapper$.toBson(CirceJsonMapper.scala:37)
  mongo4cats.circe.CirceJsonMapper$.$anonfun$toBson$2(CirceJsonMapper.scala:43)
  scala.collection.immutable.List.map(List.scala:246)
  mongo4cats.circe.CirceJsonMapper$.toBson(CirceJsonMapper.scala:43)
  mongo4cats.circe.MongoJsonCodecs.$anonfun$deriveJsonBsonValueEncoder$1(MongoJsonCodecs.scala:39)
  mongo4cats.bson.syntax$ValueSyntax$.toBson$extension(syntax.scala:26)
```

**Comment by Kirill5k** [comment]
Thanks for pointing that out. I will look into fixing this 👍

**Comment by cwholmes** [comment]
I made an attempt to fix it by inverting the cases and allowing the bigdecimal/double case to happen last so all numbers will be converted even if maybe in a less ideal fashion.

**Comment by joshcough** [comment]
Hi @Kirill5k. I see you merged a change for this. Will you be releasing it soon as well?

**Comment by Kirill5k** [comment]
> I made an attempt to fix it by inverting the cases and allowing the bigdecimal/double case to happen last so all numbers will be converted even if maybe in a less ideal fashion.

Coincidentally, I've made the fix at the same time you created the pull request.

I've released it just now. Should be available in version `0.7.1`.

**Comment by joshcough** [comment]
Amazing. Thanks.

**Comment by joshcough** [comment]
@Kirill5k we were using `0.6.13` and the new `Resource` change for starting a session (which I think went in in 0.7.0) is too much for us to take on right now. We are wondering if you could also publish 0.6.14 with these changes, so that we don't have to deal with the new Resource change just yet. Would that be possible?

**Comment by Kirill5k** [comment]
Sure, no problem, I'll release `06.14` later today

**Comment by Kirill5k** [comment]
I've released version `0.6.18` with Circe fix
